src/resolvers/Mutation/cancelMembershipRequest.ts

```diff
@@ -60,50 +56,45 @@ export const cancelMembershipRequest: MutationResolvers["cancelMembershipRequest
       );
     }
 
-    const currentUserCreatedMembershipRequest =
-      currentUser._id.toString() === membershipRequest.user.toString();
-
     // Checks whether currentUser is the creator of membershipRequest.
-    if (currentUserCreatedMembershipRequest === false) {
+    if (!(currentUser._id.toString() === membershipRequest.user.toString())) {
       throw new errors.UnauthorizedError(
         requestContext.translate(USER_NOT_AUTHORIZED_ERROR.MESSAGE),
         USER_NOT_AUTHORIZED_ERROR.CODE,
         USER_NOT_AUTHORIZED_ERROR.PARAM
       );
     }
 
-    // Deletes the membershipRequest.
-    await MembershipRequest.deleteOne({
-      _id: membershipRequest._id,
-    });
-
-    // Removes membershipRequest._id from membershipRequests list on organization's document.
-    await Organization.updateOne(
-      {
-        _id: organization._id,
-      },
-      {
-        $set: {
-          membershipRequests: organization.membershipRequests.filter(
-            (request) => request.toString() !== membershipRequest._id.toString()
-          ),
+    const promises = [
```

Comment:
instead of promises, mutations should use transactions

Reply:
I didn't get it, can you elaborate a little bit sir!?

Reply:
You mean, aggregate the queries!?

Reply:
@xoldyckk It requires the DB to be run as a replica set

Reply:
Wouldn't sending all queries at once be advantageous for connection pooling with the ORM? Your suggestion implies having one query per TCP/IP connection. Pooling would help to reduce the total number of TCP/IP connections to the API while maintaining the query count, by having one TCP/IP connection to handle groups of queries. Does the `Promise.all` make pooling possible?

Reply:
As @xoldyckk described, using the ```Promise.all``` can affect the database and atomicity wouldn't be maintained. Moreover, it can lead to database failure.

We are sending multiple database requests in Parallel using ```Promise.all``` and this can lead to database failure in case the bandwidth or throughput is low.

We are targeting small organizations too, which might depend upon small server and database instances. They might not support replica sets (cause extra capex) for fail over protection.

Reply:
@palisadoes @xoldyckk @anshgoyalevil I will it be, as it will have a different implementation from the rest of the talawa-api operation, which doesn't make any sense.